llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -13748,8 +13760,26 @@ static void PrintLoopInfo(raw_ostream &OS, ScalarEvolution *SE,
     for (const auto *P : Preds)
       P->print(OS, 4);
   }
+  Preds.clear();
 
+  auto *PredConstantMax =
+      SE->getPredicatedConstantMaxBackedgeTakenCount(L, Preds);
+  if (PredConstantMax != ConstantBTC || !Preds.empty()) {
```

Comment:
```suggestion
  if (PredConstantMax != ConstantBTC) {
    assert(!Preds.empty() && "different predicated max BTC but no predicates");
```

I *think* Preds cannot be empty if we have a different predicated max BTC?

Reply:
That makes sense, but I just used the code above that as inspiration for my change:

```
  SmallVector<const SCEVPredicate *, 4> Preds;
  auto *PBT = SE->getPredicatedBackedgeTakenCount(L, Preds);
  if (PBT != BTC || !Preds.empty()) {
```

In this case I may as well fix both cases!

Reply:
Ah, looks like that change has already been made. Another rebase on it's way then ...